### ISSUE TITLE
Rule: grouped-requires

### ISSUE THREAD

**Issue by pigulla** [eslint/eslint#214]
In the Node.JS community it is often customary to group the `require`d modules at the top of a file. Sometimes it is even wanted that they are sorted alphabetically.
##### Details

Internally, the rule should distinguish between six kinds of declarations:

``` js
var fs = require('fs'),        // "core"    \
    async = require('async'),  // "module"  |- these are "require declaration"s
    foo = require('./foo'),    // "file"    |
    bar = require('f' + 'oo'), // "unknown" /
    baz = 42,                  // "other"
    bam;                       // "uninitialized"
```

When the rule is enabled, all top-level `var` statements must satisfy the following conditions:
- either none or all variable declarations must be require declarations
- all require declarations must be of the same type
- the parameter for the call to `require` must be a simple string literal (optional)
- the module imports must be sorted alphabetically (optional)
##### Examples

The following patterns are considered okay and do not cause warnings:

``` js
// only require declarations and they are all of type "core"
var eventEmitter = require('events').EventEmitter,
    util = require('util');

// only non-require declarations
var foo = 42,
    bar = 'baz';

// this is valid if the "allowUnknown" option is enabled
var fs = require('fs'),
    foo = require('foo' + 'bar');
```

The following patterns are considered warnings:

``` js
// mixing core and module require declarations
var fs = require('fs'),
    async = require('async');

// mixing require and other declarations
var fs = require('fs'),
    i = 0;

// this is invalid if sorting is enforced
var ftp = require('ftp'),
    async = require('async');

// ...and so is this
var FooWat = require('foo/wat'),
    FooBar = require('foo/bar');
```
